File: frontend/src/types/risk.ts
export interface RiskAlert {
  type: string;
  severity: string;
  message: string;
}

export interface UserRiskSummary {
  user_id: string;
  capital: number;
  risk_tolerance_pct: number;
  trades_today: number;
  max_daily_trades: number;
  overtrading: boolean;
  open_trades: number;
  closed_trades: number;
  total_pnl: number;
  drawdown_pct: number;
  peak_equity: number;
  current_equity: number;
  alerts: RiskAlert[];
}

export interface PositionSizeResult {
  symbol: string;
  entry: number;
  stop_loss: number;
  capital: number;
  risk_tolerance_pct: number;
  risk_amount: number;
  risk_per_unit: number;
  suggested_size: number;
  max_loss: number;
  note: string;
}

File: backend/app/schemas/risk.py
from pydantic import BaseModel, Field


class PositionSizeResponse(BaseModel):
    symbol: str
    entry: float
    stop_loss: float
    capital: float
    risk_tolerance_pct: float
    risk_amount: float
    risk_per_unit: float
    suggested_size: float
    max_loss: float
    note: str


class RiskAlert(BaseModel):
    type: str
    severity: str
    message: str


class UserRiskSummary(BaseModel):
    user_id: str
    capital: float
    risk_tolerance_pct: float
    trades_today: int
    max_daily_trades: int
    overtrading: bool
    open_trades: int
    closed_trades: int
    total_pnl: float
    drawdown_pct: float
    peak_equity: float
    current_equity: float
    alerts: list[RiskAlert] = Field(default_factory=list)
